src/native/fs/fs_napi.cpp

```diff
@@ -180,17 +183,21 @@ struct FSWorker : public Napi::AsyncWorker
         , _gid(ThreadScope::orig_gid)
         , _errno(0)
         , _warn_threshold_ms(0)
+        , _took_time(0)
     {
         for (int i = 0; i < (int)info.Length(); ++i) _args_ref.Set(i, info[i]);
-        Napi::Object config = info[0].As<Napi::Object>();
-        if (config.Has("uid")) _uid = config.Get("uid").ToNumber();
-        if (config.Has("gid")) _gid = config.Get("gid").ToNumber();
-        if (config.Has("backend")) _backend = config.Get("backend").ToString();
-        if (config.Has("warn_threshold_ms")) _warn_threshold_ms = config.Get("warn_threshold_ms").ToNumber();
+        Napi::Object fs_context = info[0].As<Napi::Object>();
+        if (fs_context.Has("uid")) _uid = fs_context.Get("uid").ToNumber();
+        if (fs_context.Has("gid")) _gid = fs_context.Get("gid").ToNumber();
+        if (fs_context.Has("backend")) _backend = fs_context.Get("backend").ToString();
+        if (fs_context.Has("warn_threshold_ms")) _warn_threshold_ms = fs_context.Get("warn_threshold_ms").ToNumber();
+        if (fs_context.Has("report_fs_stats")) _report_fs_stats = Napi::Persistent(fs_context.Get("report_fs_stats").As<Napi::Function>());
+
     }
     void Begin(std::string desc)
     {
         _desc = desc;
+        _work_name = _desc.substr(0, _desc.find(" "));
```

Comment:
maybe instead of this hack we should just add work_name to the Begin function variables?

Reply:
Talking to @jackyalbo we think we don't need to add a work_name to the Begin function.